`eslint` Attempted to assign to readonly property

### What version of Bun is running?

1.0.6

### What platform is your computer?

Linux 5.15.0-86-generic x86_64 unknown

### What steps can reproduce the bug?

```
bun lint
```

### What is the expected behavior?

lint the project

### What do you see instead?

```
 bunx eslint .
  🔍 Resolving... 
Oops! Something went wrong! :(

ESLint: 8.52.0

TypeError: Attempted to assign to readonly property.
Referenced from: /app/.eslintrc.js
    at _loadPlugin (/tmp/eslint--bunx/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3441:12)
    at _loadExtendedPluginConfig (/tmp/eslint--bunx/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3228:12)
    at _loadExtends (/tmp/eslint--bunx/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3146:24)
    at _normalizeObjectConfigDataBody (/tmp/eslint--bunx/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3089:23)
    at _normalizeObjectConfigData (/tmp/eslint--bunx/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3037:20)
    at loadInDirectory (/tmp/eslint--bunx/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2878:20)
    at _loadConfigInAncestors (/tmp/eslint--bunx/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3860:10)
    at getConfigArrayForFile (/tmp/eslint--bunx/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3783:8)
    at _iterateFilesRecursive (/tmp/eslint--bunx/node_modules/eslint/lib/cli-engine/file-enumerator.js:463:45)
    at iterateFiles (/tmp/eslint--bunx/node_modules/eslint/lib/cli-engine/file-enumerator.js:300:20)
```

### Additional information

 * **file:** `.eslintrc.js`

```
module.exports = {
    parser: '@typescript-eslint/parser',
    plugins: ['@typescript-eslint'],
    extends: [
      'eslint:recommended',
      'plugin:@typescript-eslint/recommended',
    ],
    rules: {
    },
    parserOptions: {
        ecmaVersion: 2018,
        sourceType: 'module',
    },
    env: {
        browser: true,
        es6: true,
    },
};
```

 * **file:** `.eslintignore`

```
*.js
.*.js
```

I have found a solution. When I run eslint via `bunx` this is the output:

```
Attempted to assign to readonly property.
Referenced from: /workspaces/..../.eslintrc.json
```

On the other hand, by running eslint via `npx`, the output actually displays the underlying error:

```
Failed to load plugin 'prettier' declared in '.eslintrc.json': Cannot find module 'eslint-plugin-prettier'
Require stack:
- /Users/lucadibello/Projects/beer-search-engine/beer-search-engine/__placeholder__.js
Referenced from: /Users/lucadibello/Projects/beer-search-engine/beer-search-engine/.eslintrc.json
```

I simply just forgot to install the package `eslint-plugin-prettier`.

--- a/src/bun.js/resolve_message.classes.ts
+++ b/src/bun.js/resolve_message.classes.ts
@@ -12,6 +12,7 @@ export default [
       message: {
         getter: "getMessage",
         cache: true,
+        writable: true,
       },
       code: {
         getter: "getCode",
@@ -66,6 +67,7 @@ export default [
       message: {
         getter: "getMessage",
         cache: true,
+        writable: true,
       },
       name: {
         value: "BuildMessage",

--- a/src/codegen/class-definitions.ts
+++ b/src/codegen/class-definitions.ts
@@ -10,7 +10,15 @@ interface PropertyAttribute {
 }
 
 export type Field =
-  | ({ getter: string; cache?: true | string; this?: boolean } & PropertyAttribute)
+  | ({
+      getter: string;
+      cache?: true | string;
+      /**
+       * Allow overriding the value of the property
+       */
+      writable?: boolean;
+      this?: boolean;
+    } & PropertyAttribute)
   | { value: string }
   | ({ setter: string; this?: boolean } & PropertyAttribute)
   | ({

--- a/src/codegen/generate-classes.ts
+++ b/src/codegen/generate-classes.ts
@@ -187,6 +187,7 @@ function propRow(
     configurable = false,
     value,
     builtin,
+    writable = false,
   } = (defaultPropertyAttributes ? Object.assign({}, defaultPropertyAttributes, prop) : prop) as any;
 
   var extraPropertyAttributes = "";
@@ -209,7 +210,7 @@ function propRow(
     if (getter) {
       getter = symbol + "GetterWrap";
     }
-    if (setter) {
+    if (setter || writable) {
       setter = symbol + "SetterWrap";
     }
     if (fn) {
@@ -219,7 +220,7 @@ function propRow(
     if (getter) {
       getter = symbolName(typeName, getter);
     }
-    if (setter) {
+    if (setter || writable) {
       setter = symbolName(typeName, setter);
     }
     if (fn) {
@@ -252,8 +253,11 @@ function propRow(
 { "${name}"_s, static_cast<unsigned>(JSC::PropertyAttribute::CustomAccessor | JSC::PropertyAttribute::DOMAttribute${extraPropertyAttributes}), NoIntrinsic, { HashTableValue::GetterSetterType, ${getter}, ${setter} } }
 `.trim();
   } else if (defaultValue) {
-  } else if (getter && !supportsObjectCreate) {
+  } else if (getter && !supportsObjectCreate && !writable) {
     return `{ "${name}"_s, static_cast<unsigned>(JSC::PropertyAttribute::ReadOnly | JSC::PropertyAttribute::CustomAccessor | JSC::PropertyAttribute::DOMAttribute${extraPropertyAttributes}), NoIntrinsic, { HashTableValue::GetterSetterType, ${getter}, 0 } }
+`.trim();
+  } else if (getter && !supportsObjectCreate && writable) {
+    return `{ "${name}"_s, static_cast<unsigned>(JSC::PropertyAttribute::CustomAccessor | JSC::PropertyAttribute::DOMAttribute${extraPropertyAttributes}), NoIntrinsic, { HashTableValue::GetterSetterType, ${getter}, ${setter} } } 
 `.trim();
   } else if (getter && supportsObjectCreate) {
     setter = getter.replace("Get", "Set");
@@ -745,6 +749,7 @@ function renderDecls(symbolName, typeName, proto, supportsObjectCreate = false) 
         } JSC::JSGlobalObject* lexicalGlobalObject);`,
         `
     JSC_DECLARE_CUSTOM_GETTER(${symbolName(typeName, name)}GetterWrap);
+    ${proto[name].writable ? `JSC_DECLARE_CUSTOM_SETTER(${symbolName(typeName, name)}SetterWrap);` : ""}
     `.trim(),
         "\n",
       );
@@ -903,6 +908,18 @@ JSC_DEFINE_CUSTOM_GETTER(${symbolName(typeName, name)}GetterWrap, (JSGlobalObjec
     thisObject->${cacheName}.set(vm, thisObject, result);
     RELEASE_AND_RETURN(throwScope, JSValue::encode(result));
 }`);
+          if (proto[name].writable) {
+            rows.push(`
+            JSC_DEFINE_CUSTOM_SETTER(${symbolName(typeName, name)}SetterWrap, (JSGlobalObject * lexicalGlobalObject, EncodedJSValue encodedThisValue, EncodedJSValue encodedValue, PropertyName attributeName))
+            {
+                auto& vm = lexicalGlobalObject->vm();
+                auto throwScope = DECLARE_THROW_SCOPE(vm);
+                ${className(typeName)}* thisObject = jsCast<${className(typeName)}*>(JSValue::decode(encodedThisValue));
+                JSC::EnsureStillAliveScope thisArg = JSC::EnsureStillAliveScope(thisObject);
+                thisObject->${cacheName}.set(vm, thisObject, JSValue::decode(encodedValue));
+                RELEASE_AND_RETURN(throwScope, true);
+            }`);
+          }
         } else {
           rows.push(`
           JSC_DEFINE_CUSTOM_GETTER(${symbolName(typeName, name)}GetterWrap, (JSGlobalObject * globalObject, EncodedJSValue encodedThisValue, PropertyName attributeName))

--- a/test/js/bun/resolve/build-error.test.ts
+++ b/test/js/bun/resolve/build-error.test.ts
@@ -1,9 +1,12 @@
 test("BuildError is modifiable", async () => {
   try {
-    await import("./inspect-error-fixture-bad.js");
+    await import("../util/inspect-error-fixture-bad.js");
     expect.unreachable();
   } catch (e) {
     var error: BuildMessage = e as BuildMessage;
+    if (error.name !== "BuildMessage") {
+      throw new Error("Expected BuildMessage, got " + error.name);
+    }
   }
 
   const message = error!.message;

--- a/test/js/bun/resolve/build-error.test.ts
+++ b/test/js/bun/resolve/build-error.test.ts
@@ -1,0 +1,14 @@
+test("BuildError is modifiable", async () => {
+  try {
+    await import("./inspect-error-fixture-bad.js");
+    expect.unreachable();
+  } catch (e) {
+    var error: BuildMessage = e as BuildMessage;
+  }
+
+  const message = error!.message;
+  // @ts-ignore
+  expect(() => (error!.message = "new message")).not.toThrow();
+  expect(error!.message).toBe("new message");
+  expect(error!.message).not.toBe(message);
+});

--- a/test/js/bun/resolve/resolve-error.test.ts
+++ b/test/js/bun/resolve/resolve-error.test.ts
@@ -9,6 +9,17 @@ describe("ResolveMessage", () => {
     }
   });
 
+  it(".message is modifiable", async () => {
+    try {
+      await import("./file-importing-nonexistent-file.js");
+    } catch (e: any) {
+      const orig = e.message;
+      expect(() => (e.message = "new message")).not.toThrow();
+      expect(e.message).toBe("new message");
+      expect(e.message).not.toBe(orig);
+    }
+  });
+
   it("has code for esm", async () => {
     try {
       await import("./file-importing-nonexistent-file.js");

